src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -37,7 +55,28 @@ public void process(Message<ChsDelta> chsDelta) {
             //TODO We need to create a new one for this processor
             final boolean isRetry = headers.containsKey("INSOLVENCY_DELTA_RETRY_COUNT");
             final ChsDelta payload = chsDelta.getPayload();
+            final String logContext = payload.getContextId();
+            final Map<String, Object> logMap = new HashMap<>();
+
+            ObjectMapper mapper = new ObjectMapper();
+            InsolvencyDelta insolvencyDelta = mapper.readValue(payload.getData(),
+                    InsolvencyDelta.class);
+            logger.trace(String.format("InsolvencyDelta extracted "
+                    + "from Kafka message: %s", insolvencyDelta));
 
+            // We always receive only one insolvency per delta from CHIPS
+            Insolvency insolvency = insolvencyDelta.getInsolvency().get(0);
+            final String companyNumber = insolvency.getCompanyNumber();
+
+            final ApiResponse<CompanyProfile> response =
+                    companyProfileService.getCompanyProfile(logContext, companyNumber);
+            logger.trace(String.format("Retrieved company profile for company number %s: %s",
+                    companyNumber, response.getData()));
+            handleResponse(HttpStatus.valueOf(response.getStatusCode()), logContext,
+                    "Response from GET call to company profile api", logMap, logger);
+
+            // TODO check if company needs updating - use response.getData()
+            // TODO PATCH company profile
```

Comment:
You can mention the ticket number in the TODO (DSND-603)

Reply:
Good shout, done